www/js/history-widget.js

```diff
@@ -117,38 +117,47 @@ define([
 		loadHandler : function(e){
 			var label = e.target.id;
 			var session_id = label.slice(4,label.indexOf('_label'));
-			//before attaching the session id or x parameter
-			//verify if it is undefined , then add form property
-			//else just modify the value with current session id depending on history item opted
-			var form = document.forms[formID];
-			if(form.x === undefined){
-				addFormProperty("x",session_id);
-			}
-			else{
-				form.x.value = session_id;
-			}
-			submit();		
+			console.log("session_id",session_id);
+			//history_form has already been created and attached in initial handlers
+			var form = document.forms["history_form"];
+			//update the session id parameter based on the particular session in history
+			//to be loaded identified by session_id
+			form.x.value = session_id;
+			form.submit();		
 		},
 		initHandlers : function(){
 			var sessions = this.sessions;
 			var query_params = this.params;
 
-			//Before attaching click handlers to buttons which open each session
-			//create a form which reads parameters from session
-			//we already have js file for this purpose : form.js
-			//step 1: create a form and add properties from parameters
-			console.log("form id is",formID);
-			createForm();
+			//create an array of form element objects			
+			//This form is hidden and will be used to load sessions using session ids
+			var elements_data = [];
+			var session_element_exists = false;
 			for (var prop in query_params) {
- 				addFormProperty(prop,query_params[prop]);
+				if(prop == "x") session_element_exists = true;
+				elements_data.push({
+					name: prop,
+					type: "text",
+					value: query_params[prop]
+				})
+			}
+			if(!session_element_exists){
+				elements_data.push({
+					name: "x",
+					type: "text",
+					value: ""
+				})
```

Comment:
we should follow one rule, whether to put semi colon at the end or not. I think since we are putting it would it be possible to just to stick to that, just to maintain uniformity.